client/my-sites/media-library/content.jsx

```diff
@@ -247,6 +295,10 @@ const MediaLibraryContent = React.createClass( {
 
 export default connect( ( state, ownProps ) => {
 	return {
-		siteSlug: ownProps.site ? getSiteSlug( state, ownProps.site.ID ) : ''
+		siteSlug: ownProps.site ? getSiteSlug( state, ownProps.site.ID ) : '',
+		connectedServices: toArray( getKeyringConnections( state ) ).filter( item => item.type === 'other' && item.status === 'ok' ),
```

Comment:
Keyring connections are split into 'publicize' and 'other', with google photos belonging to 'other'. The API request returns all keyring connections, and this bit of code filters out the 'other' ones, which is all we are interested in - we end up with an array of these other connections.

We then use this array with `some( this.props.connectedServices, item => item.service === this.props.source )` (a bit earlier in the code) to filter it further down to the particular data source used in the modal.

If extra services are added to 'other' then it won't make any difference as we only look for the one used by the data source.